components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -116,6 +116,117 @@ std::string URLMethodToRequestType(ledger::type::UrlMethod method) {
   }
 }
 
+bool ResetOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::vector<std::string>& paths) {
+  bool result = true;
+
+  for (const auto& path : paths) {
+    const base::FilePath file_path =
+        base::FilePath::FromUTF8Unsafe(path.c_str());
+    if (!base::DeletePathRecursively(file_path)) {
+      result = false;
+    }
+  }
+
+  if (diagnostic_log) {
+    diagnostic_log->Close();
```

Comment:
It's surprising that we first delete the log file (it's in the paths vector, correct?) before we call `Close()` on it. Is it possible to move this up before we delete paths?

Reply:
You're right, that was a last minute change I made. Will fix.